test/test_epoll.cpp

```diff
@@ -555,3 +556,212 @@ TEST(CEPoll, ThreadedUpdate)
+
+            int wlen = 2;
+            SRTSOCKET write[2];
+
+            ASSERT_NE(srt_epoll_wait(m_server_pollid,
+                        read,  &rlen,
+                        write, &wlen,
+                        -1, // -1 is set for debuging purpose.
+                        // in case of production we need to set appropriate value
+                        0, 0, 0, 0), SRT_ERROR );
+
+            ASSERT_EQ(rlen, 1); // get exactly one read event without writes
+            ASSERT_EQ(wlen, 0); // get exactly one read event without writes
+            ASSERT_EQ(read[0], servsock); // read event is for bind socket    	
+        }
+
+        sockaddr_in scl;
+        int sclen = sizeof scl;
+
+        SRTSOCKET sock = srt_accept(servsock, (sockaddr*)&scl, &sclen);
+        ASSERT_NE(sock, SRT_INVALID_SOCK);
+
+        srt_epoll_add_usock(m_server_pollid, sock, &epoll_in); // wait for input
+
+        { // wait for 1316 packet from client
+            int rlen = 2;
+            SRTSOCKET read[2];
+
+            int wlen = 2;
+            SRTSOCKET write[2];
+
+            ASSERT_NE(srt_epoll_wait(m_server_pollid,
+                        read,  &rlen,
+                        write, &wlen,
+                        -1, // -1 is set for debuging purpose.
+                        // in case of production we need to set appropriate value
+                        0, 0, 0, 0), SRT_ERROR );
+
+            ASSERT_EQ(rlen, 1); // get exactly one read event without writes
+            ASSERT_EQ(wlen, 0); // get exactly one read event without writes
+            ASSERT_EQ(read[0], sock); // read event is for bind socket        
+        }
+
+        char buffer[1316];
+        ASSERT_EQ(srt_recvmsg(sock, buffer, sizeof buffer), 1316);
+
+        // char pattern[4] = {1, 2, 3, 4};
+        // EXPECT_TRUE(ArraysEqual(pattern, buffer, sizeof pattern));
+
+        std::cout << "serverSocket waiting..." << std::endl;
+        {
+            int rlen = 2;
+            SRTSOCKET read[2];
+
+            int wlen = 2;
+            SRTSOCKET write[2];
+
+            ASSERT_EQ(-1, srt_epoll_wait(m_server_pollid,
+                        read,  &rlen,
+                        write, &wlen,
+                        2000,
+                        0, 0, 0, 0));
+            const int last_error = srt_getlasterror(NULL);
+            ASSERT_EQ(SRT_ETIMEOUT, last_error) << last_error;
+        }
+        std::cout << "serverSocket finished waiting" << std::endl;
+
+        srt_close(sock);
+        srt_close(servsock);
+        srt_close(m_client_sock); // cannot close m_client_sock after srt_sendmsg because of issue in api.c:2346 
+
+        client.join();
+    }
+
+};
+
+
+TEST_F(TestEPoll, SimpleAsync)
+{
+	ASSERT_EQ(srt_startup(), 0);
+
+    m_client_pollid = srt_epoll_create();
+    ASSERT_NE(SRT_ERROR, m_client_pollid);
+
+    m_server_pollid = srt_epoll_create();
+    ASSERT_NE(SRT_ERROR, m_server_pollid);
+
+    std::thread server( [this] {serverSocket(); });
```

Comment:
Yeah, starting a thread and joining it immediately thereafter definitely doesn't make any sense, especially if you do completely nothing in the main thread.

The function's name is good, though. It does start a client from inside itself, that's true, but aside of this it does exactly what it is predicted to do. It starts the client from itself because the socket needs to be set up before the client can start - otherwise it would have to be split into two separate functions. I don't think it's so important when it's just a unit test.

Having the pollids in the main function instead of their functions isolates them from the test procedures so that they can work freely with a prediction that these eids simply always exist and they don't have to worry about any setup/teardown for them.